apis/v1alpha2/validation/httproute.go

```diff
@@ -40,7 +40,11 @@ func ValidateHTTPRoute(route *gatewayv1a2.HTTPRoute) field.ErrorList {
 // validateHTTPRouteSpec validates that required fields of spec are set according to the
 // HTTPRoute specification.
 func validateHTTPRouteSpec(spec *gatewayv1a2.HTTPRouteSpec, path *field.Path) field.ErrorList {
-	return validateHTTPRouteUniqueFilters(spec.Rules, path.Child("rules"))
+	if errList := validateHTTPRouteUniqueFilters(spec.Rules, path.Child("rules")); len(errList) > 0 {
+		return errList
+	}
+
+	return nil
```

Comment:
Had to double check this, but it looks like we return empty error lists instead of nil in upstream (https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L2723)

```suggestion
	return validateHTTPRouteUniqueFilters(spec.Rules, path.Child("rules"))
```